FAIL: Bug: next_delay_ns already set for timer with SGXLKL_ETHREADS=1

With SGXLKL_ETHREADS=1 below 4 tests fails in nightly sgx-lkl pipeline:

[[ SGX-LKL ]] FAIL: Bug: next_delay_ns already set for timer
[ SGX-LKL ] init (0: 19 exit=0) [ SGX-LKL ] FAIL: sgxlkl_ethread_init() failed (ethread_id=0 result=19 (OE_ENCLAVE_ABORTING))

**These tests fail with same failure**:
basic-sleep-(debug)-(run-hw)-(1-ethreads)
basic-nonroot_halt_test-(debug)-(run-hw)-(1-ethreads)
java-thread-(debug)-(run-hw)-(1-ethreads) 
dotnet-(debug)-(run-hw)-(1-ethreads)

@prp repro steps:
1. Build sgx-lkl
2. Go under /sgx-lkl/tests/basic/sleep
3. Modify Makefile and add SGXLKL_ETHREADS=1 to the beginning of SGXLKL_ENV
4. Run "make run-hw"
5. This doesn't fail every time. But if you run 5-10 times you will hit the bug.  

Here is a sample log from local run when failed:
[[  SGX-LKL ]] lkl_start_init(): lkl_start_kernel() finished
[[  SGX-LKL ]] lkl_start_init(): creating LKL termination thread
[[  SGX-LKL ]] create_lkl_termination_thread(): enter
[[  SGX-LKL ]] lkl_start_init(): calling lkl_mount_virtial()
[[  SGX-LKL ]] lkl_termination_thread(): enter
[[  SGX-LKL ]] lkl_termination_thread(): Performed LKL syscall to get host task allocated (pid=36)
[[  SGX-LKL ]] lkl_start_init(): calling init_random()
[[  SGX-LKL ]] init_random(): Adding entropy to entropy pool
[    0.096381] random: crng init done
[[  SGX-LKL ]] init_enclave_clock(): Setting enclave realtime clock
[[  SGX-LKL ]] wg0 has public key foHEDdgtbg7TJoPDsoONHRov25XpLK9aqUAEzzvCbAM=
[    0.098646] virtio_blk virtio1: [vda] 10240 512-byte logical blocks (5.24 MB/5.00 MiB)
[[  SGX-LKL ]] FAIL: Bug: next_delay_ns already set for timer
2020-05-28T18:12:54.000000Z [(H)ERROR] tid(0x7f9c2076e700) | :OE_ENCLAVE_ABORTING [../host/calls.c:oe_call_enclave_function_by_table_id:91]
[   SGX-LKL  ] init (0: 19 exit=0) [   SGX-LKL  ] FAIL: sgxlkl_ethread_init() failed (ethread_id=0 result=19 (OE_ENCLAVE_ABORTING))
Makefile:29: recipe for target 'run-hw' failed
make: *** [run-hw] Error 1

--- a/src/lkl/posix-host.c
+++ b/src/lkl/posix-host.c
@@ -520,6 +520,7 @@ static void* timer_callback(void* _timer)
         // Check if the timer has triggered
         if (did_timeout)
         {
+            timer->next_delay_ns = 0;
             timer->callback_fn(timer->callback_arg);
             // If the callback function itself resets the timer,
             // timer->next_delay_ns will be non-zero.

